src/components/AlignGroup.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  useLayoutEffect,
+  useRef,
+  useMemo,
+  useState,
+  ReactNode,
+} from 'react';
+
+type Align = 'start' | 'middle' | 'end';
+
+export interface AlignGroupProps {
+  x?: number;
+  y?: number;
+  verticalAlign?: Align;
+  horizontalAlign?: Align;
+  children: ReactNode | ReactNode[];
+}
+
+function calculatePosition(start: number, align: Align, space: number) {
+  switch (align) {
+    case 'start': {
+      return start;
+    }
+    case 'end': {
+      return start - space;
+    }
+    case 'middle': {
+      return start - space / 2;
+    }
+    default: {
+      throw new Error(`Unkwnown alignment ${JSON.stringify(align)}`);
+    }
+  }
+}
+
+export function AlignGroup(props: AlignGroupProps) {
+  const {
+    x = 0,
+    y = 0,
+    verticalAlign = 'start',
+    horizontalAlign = 'start',
+    children,
+  } = props;
+
+  const groupRef = useRef<SVGGElement | null>(null);
```

Comment:
```suggestion
  const groupRef = useRef<SVGGElement>(null);
```

The  `| null` is implied by passing `null` as an initial value